### ISSUE TITLE
Build Neovim from source when Nightly build fails

### ISSUE THREAD

**Issue by rhysd** [rhysd/action-setup-vim#9]
## Problem

action-setup-vim fails when Nightly Neovim release is broken. Example of job failure:

https://github.com/rhysd/action-setup-vim/runs/1313461451?check_suite_focus=true

It failed because Nightly Neovim release did not provide macOS package due to failure of Neovim CI.

## Solution

To solve the issue, I think providing fallback would be better. When downloading asset fails, this action can build the latest Neovim from source.

Related to #7

**Comment by kristijanhusak** [comment]
Just a note that there is similar issue with Windows now.

https://github.com/kristijanhusak/vim-dadbod-ui/runs/1588245067?check_suite_focus=true

**Comment by rhysd** [comment]
Yes. Neovim omitted Windows build from nightly though I don't know it is intended or accidental. So there is no nightly Windows asset in the release.

https://github.com/neovim/neovim/releases/tag/nightly

**Comment by voldikss** [comment]
This feature is quite necessary. Just note that neovim's nightly build was broken today, the action removed the nightly tag but failed to create a new one. https://github.com/neovim/neovim/pull/13732

**Comment by voldikss** [comment]
May I ask: if I want to test on neovim stable and nightly build and vim nightly build only, how could I write the yml for?

**Comment by rhysd** [comment]
Hi @voldikss, I'm doubtful about this feature being useful actually. When nightly build is broken, it means that Neovim can't be built at the moment. If this action falls back to building Neovim from source, the build would fail. The feature is useful only when (1) nightly build failed && (2) the build was fixed by someone && (3) next nightly build is not triggered yet.

> May I ask: if I want to test on neovim stable and nightly build and vim nightly build only, how could I write the yml for?

My recommendation is using `exclude`:

```yaml
strategy:
  matrix:
    neovim: [true, false]
    version: [nightly, stable]
    exclude:
      - neovim: false
        version: stable
```

**Comment by voldikss** [comment]
>  If this action falls back to building Neovim from source, the build would fail

Right, I agree with you. Build from souce seems more a unnecessary duplication of neovim's own build.

> My recommendation is using exclude:

Thank you for the reply!

**Comment by rhysd** [comment]
This was solved by #18. Closing.
